StringCalculatorKata/StringCalculator.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace StringCalculatorKata
+{
+    public static class StringCalculator
+    {
+        public static int Add(string numbers)
+        {
+            var values = new List<string>();
+
+            if (numbers.StartsWith("//"))
+            {
+                values = GetValuesWithCustomDelimiter(numbers);
+            }
+            else
+            {
+                values = numbers.Split(',', '\n')
+                                .ToList();
+            }
+
+            var intNumbers = GetIntegers(values);
+
+            if (HasNegatives(intNumbers))
+            {
+                throw new NegativesNotAllowedException(GetNegatives(intNumbers));
+            }
+
+            return SumOfIntegers(intNumbers);
+        }
+
+        private static List<int> GetIntegers(List<string> values)
+        {
+            var intNumbers = new List<int>();
+            int tempInteger;
+
+            for (int i = 0; i < values.Count; i++)
+            {
+                int.TryParse(values[i], out tempInteger);
+                intNumbers.Add(tempInteger);
+            }
+
+            return intNumbers;
+        }
+
+        private static bool HasNegatives(List<int> intNumbers)
+        {
+            foreach (var number in intNumbers)
+            {
+                if (number < 0)
+                    return true;
+            }
+
+            return false;
+        }
+
+        private static List<int> GetNegatives(List<int> intNumbers)
```

Comment:
u can return instead:
`intNumbers.Where(number => number < 0).ToList();
`
don't forget that u learned LINQ :D u can use it instead of boring loops

Reply:
Done